src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -19,16 +24,32 @@
 
     private final BlobServiceClient storageClient;
     private final BlobDispatcher dispatcher;
+    private final EnvelopeRepository envelopeRepository;
 
     public BlobProcessor(
         BlobServiceClient storageClient,
-        BlobDispatcher dispatcher
+        BlobDispatcher dispatcher,
+        EnvelopeRepository envelopeRepository
     ) {
         this.storageClient = storageClient;
         this.dispatcher = dispatcher;
+        this.envelopeRepository = envelopeRepository;
     }
 
     public void process(String blobName, String containerName) {
+        envelopeRepository.find(blobName, containerName).ifPresentOrElse(
+            envelope -> logger.warn(
+                "Envelope already processed in system. ID: {}, filename: {}, container: {}, state: {}",
```

Comment:
worth adding info on what we do about. ie, skipping processing.

Reply:
Not even sure if it should be a warning - this will happen from time to time, because deletion job will be a bit behind.

Reply:
i would vote for debug, but please propose what we would want?

Reply:
info